sdk/core/core/inc/az_context.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_context.h
+ *
+ * @brief az_context for cancelling execution
+ */
+
+#ifndef _az_CONTEXT_H
+#define _az_CONTEXT_H
+
+#include <az_result.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_context az_context;
+
+/**
+ * @brief An az_context value in a node in a tree that represents expiration times and key/value
+ * pairs. The root node in the tree (ultimate parent) is az_context_app which is a context for the
+ * entire application. Each new node is a child of some parent.
+ */
+struct az_context {
+  struct {
+    az_context const * parent; // Pointer to parent context (or NULL); immutable after creation
+    int64_t expiration; // Time when context expires
+    void *key, *value; // Pointers to the key & value (usually NULL)
```

Comment:
Apply the new formatting style, and I also recommend not to use multiple variable declarations.
```
void* key;
void* value;
```